test/backend_nnp_datatype_test.in.cpp

```diff
@@ -0,0 +1,217 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <algorithm>
+#include <algorithm>
+#include <cinttypes>
+#include <cmath>
+#include <cstdlib>
+#include <random>
+#include <string>
+#include "gtest/gtest.h"
+
+#include "ngraph/autodiff/adjoints.hpp"
+#include "ngraph/graph_util.hpp"
+#include "ngraph/log.hpp"
+#include "ngraph/ngraph.hpp"
+#include "ngraph/op/experimental/generate_mask.hpp"
+#include "ngraph/serializer.hpp"
+#include "ngraph/state/rng_state.hpp"
+#include "util/all_close.hpp"
+#include "util/all_close_f.hpp"
+#include "util/ndarray.hpp"
+#include "util/random.hpp"
+#include "util/test_control.hpp"
+#include "util/test_tools.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+
+NGRAPH_TEST(${BACKEND_NAME}, scalar_constant_int64_bfloat_fail)
+{
+    auto r = op::Constant::create(element::i64, Shape{}, {501});
```

Comment:
Just as with int32 you should be using numbers that exercise int64. 501 can't fit in a bfloat but it can fit in a float or an int32. The numbers you test should only pass if they are implemented as 64 bit integers internally.
The same argument holds for the other tests.